### PR TITLE
Preparing for new AKS AAT environment

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5368

### Change description ###

Preparing for new AAT AKS environment.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: charts/cmc-claim-store/requirements.yaml [hunk 1]
```diff
@@ -3,11 +3,11 @@ dependencies:
     version: ~2.6.5
     repository: '@stable'
   - name: java
-    version: '>1.0.0'
+    version: ~1.0.7
```
Comment (timja) [inline charts/cmc-claim-store/requirements.yaml]:
```suggestion
    version: ~1.1.0
```

File: charts/cmc-claim-store/requirements.yaml [hunk 2]
```diff
@@ -3,11 +3,11 @@ dependencies:
...
     repository: '@hmcts'
     alias: claim-store
   - name: nodejs
-    version: 0.0.11
+    version: 0.1.4
```
Comment (timja) [inline charts/cmc-claim-store/requirements.yaml]:
```suggestion
    version: 0.2.0
```

File: charts/cmc-claim-store/requirements.yaml [hunk 3]
```diff
@@ -3,11 +3,11 @@ dependencies:
...
     repository: '@hmcts'
   - name: rpe-feature-toggle-api
```
Comment (timja) [inline charts/cmc-claim-store/requirements.yaml]: this should be conditional based on a tag, see draft store or user-profile api

File: src/main/resources/bootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      prefixed: false
+      aliases:
+        claim-store-s2s-secret: IDAM_S2S_AUTH_TOTP_SECRET
+        notify-api-key: GOV_NOTIFY_API_KEY
+        citizen-oauth-client-secret: oauth2.client.secret
+        anonymous-caseworker-username: IDAM_CASEWORKER_ANONYMOUS_USERNAME
+        anonymous-caseworker-password: IDAM_CASEWORKER_ANONYMOUS_PASSWORD
+        system-update-username: IDAM_CASEWORKER_SYSTEM_USERNAME
+        system-update-password: IDAM_CASEWORKER_SYSTEM_PASSWORD
+        AppInsightsInstrumentationKey: APPINSIGHTS_INSTRUMENTATIONKEY
```
Comment (timja) [inline src/main/resources/bootstrap.yml]: this is the wrong property on java logging 5.0.1
https://github.com/hmcts/java-logging/tree/master/java-logging-appinsights#provide-instrumentation-key

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/notifications/BaseNotificationServiceTest.java [hunk 5]
```diff
@@ -32,11 +37,10 @@
     protected static final String USER_EMAIL = "user@example.com";
     protected static final String USER_FULLNAME = "Steven Patrick";
 
-    private PrintStream systemOut;
-    private ByteArrayOutputStream outContent;
-
     protected final Claim claim = SampleClaim.getDefault();
 
+    protected Logger log = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
```
Comment (timja) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/notifications/BaseNotificationServiceTest.java]: Are these changes related to the updates?

File: charts/cmc-claim-store/requirements.yaml [hunk 6]
```diff
@@ -3,11 +3,11 @@ dependencies:
     version: ~2.6.5
     repository: '@stable'
```
Comment (timja) [inline charts/cmc-claim-store/requirements.yaml]: this should also be conditional based on a tag

File: Jenkinsfile_CNP [hunk 7]
```diff
@@ -66,5 +52,4 @@ withPipeline("java", product, component) {
   enableSlackNotifications(notificationsChannel)
   enableDockerBuild()
   installCharts()
-  setVaultName('cmc')
```
Comment (timja) [inline Jenkinsfile_CNP]: you should probably also stop outputting the vault name as part of this change:
https://github.com/hmcts/cmc-claim-store/blob/master/infrastructure/output.tf#L1-L7

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/notifications/BaseNotificationServiceTest.java (hunk 5 above)
Comment (markry-kainos) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/notifications/BaseNotificationServiceTest.java]: yep broken with logging lib update

File: charts/cmc-claim-store/requirements.yaml (hunk 3 above)
Comment (markry-kainos) [reply charts/cmc-claim-store/requirements.yaml]: yep, like it - have been meaning to use for postgres to use 1 instance :)

File: charts/cmc-claim-store/requirements.yaml [hunk 8]
```diff
@@ -2,13 +2,17 @@ dependencies:
...
+    tags:
+      - postgresql-pod
   - name: java
-    version: '>1.0.0'
+    version: ~1.1.0
     repository: '@hmcts'
     alias: claim-store
   - name: nodejs
```
Comment (timja) [inline charts/cmc-claim-store/requirements.yaml]: should this even be here?

**Comment by markry-kainos** [comment]
@timja green build - i will merge before pointing to new cluster

File: src/main/resources/bootstrap.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+      paths: /mnt/secrets/cmc
+      prefixed: false
+      aliases:
+        claim-store-s2s-secret: IDAM_S2S_AUTH_TOTP_SECRET
+        notify-api-key: GOV_NOTIFY_API_KEY
+        citizen-oauth-client-secret: oauth2.client.secret
+        anonymous-caseworker-username: IDAM_CASEWORKER_ANONYMOUS_USERNAME
+        anonymous-caseworker-password: IDAM_CASEWORKER_ANONYMOUS_PASSWORD
+        system-update-username: IDAM_CASEWORKER_SYSTEM_USERNAME
+        system-update-password: IDAM_CASEWORKER_SYSTEM_PASSWORD
```
Comment (adusumillipraveen) [inline src/main/resources/bootstrap.yml]: You will have to add alias mapping for passwords in here. They won't impact till you add secrets to mount .

Comment (adusumillipraveen) [reply src/main/resources/bootstrap.yml]: I wonder if spring will automatically relax bind claim-store-db-password to CLAIM_STORE_DB_PASSWORD . it might do it to claimStoreDbPassword.

File: charts/cmc-claim-store/values.template.yaml [hunk 10]
```diff
@@ -8,127 +12,94 @@ postgresql:
...
+java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
   applicationPort: 4400
   cpuRequests: 25m
+  keyVaults:
+    cmc:
+      resourceGroup: cmc
+      secrets:
+        - AppInsightsInstrumentationKey
```
[the remark this answers is not in the document]
Comment (adusumillipraveen) [reply charts/cmc-claim-store/values.template.yaml]: We might not need this here , unless you want it in preview

File: src/main/resources/bootstrap.yml (hunk 9 above)
Comment (markry-kainos) [reply src/main/resources/bootstrap.yml]: this works though 🤔 green build with smoke/functional tests passing - so spring is binding db password (and any other secrets that match) ok. what do you think? does something still need doing? :)

File: charts/cmc-claim-store/values.template.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (markry-kainos) [reply charts/cmc-claim-store/values.template.yaml]: Agreed - we prob dont need unless app wont start without it! Ill check 👍

File: src/main/resources/bootstrap.yml (hunk 9 above)
Comment (adusumillipraveen) [reply src/main/resources/bootstrap.yml]: It might be working because you are not taking passwords from secrets on preview- they are coming from values.aat.teamplte.yaml.
